### PR TITLE
CertificateService refactoring

### PR DESCRIPTION
**Pull request by lorthirk**
Hi all,

The aim of this PR is to make CertificateService following the KapuaService blueprint, so that now reusing it should be easier and more in a Kapua style. As a result, JWT tokens are not signed with a public and a private key but rather with a self-signed certificate and a private key.

Cheers

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <!--
-    Copyright (c) 2011, 2017 Eurotech and/or its affiliates and others
+    Copyright (c) 2017 Eurotech and/or its affiliates and others
```
Comment (Coduz) [inline pom.xml]: isn't thins supposed to be 2016-2017?

File: service/security/certificate/internal/src/main/java/org/eclipse/kapua/service/certificate/internal/CertificateServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    private static final CertificateFactory CERTIFICATE_FACTORY= LOCATOR.getFactory(CertificateFactory.class);
+    private static final PermissionFactory PERMISSION_FACTORY = LOCATOR.getFactory(PermissionFactory.class);
+    private static final CertificateDomain CERTIFICATE_DOMAIN = new CertificateDomain();
+
+    /**
+     * Constructor
+     */
+    public CertificateServiceImpl() throws KapuaException {
+        KapuaSecurityUtils.doPrivileged(() -> {
+            AUTHORIZATION_SERVICE.checkPermission(PERMISSION_FACTORY.newPermission(CERTIFICATE_DOMAIN, Actions.write, KapuaId.ANY));
```
Comment (Coduz) [inline service/security/certificate/internal/src/main/java/org/eclipse/kapua/service/certificate/internal/CertificateServiceImpl.java]: Uhmmm calling `KapuaSecurityUtils.doPrivileged(()` and then calling `AUTHORIZATION_SERVICE.checkPermission(...)` seems strange :) Probably the `.checkAccess(...)` should be moved to the `.query(...)` method which currently is missing the `checkAccess` and the parameter validation.

File: service/security/certificate/internal/src/main/java/org/eclipse/kapua/service/certificate/internal/CertificateServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+    }
+
+    @Override
+    public Certificate find(KapuaId scopeId, KapuaId entityId) throws KapuaException {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public CertificateListResult query(KapuaQuery<Certificate> query) throws KapuaException {
+        Certificate kapuaCertificate = new CertificateImpl(query.getScopeId());
```
Comment (Coduz) [inline service/security/certificate/internal/src/main/java/org/eclipse/kapua/service/certificate/internal/CertificateServiceImpl.java]: Please add `ArgumentValidator` checks for `null` query and `null` query.getScopeId().
Please add the `AUTHORIZATION_SERVICE.checkPermission(...)` to check that the user has `certificate:view` permission

File: service/security/certificate/internal/src/main/resources/liquibase/0.3.x/changelog-certificate-0.3.x.xml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+    Contributors:
+        Eurotech - initial API and implementation
+ -->
+<databaseChangeLog
+        xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog
+                      http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.0.xsd"
+        logicalFilePath="KapuaDB/changelog-certificate-0.3.x.xml">
```
Comment (Coduz) [inline service/security/certificate/internal/src/main/resources/liquibase/0.3.x/changelog-certificate-0.3.x.xml]: Version should be 0.4.0

File: service/security/certificate/internal/src/main/resources/liquibase/0.3.x/crt_certificate.xml [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+    Contributors:
+        Eurotech - initial API and implementation
+ -->
+<databaseChangeLog
+  xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog
+                      http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.0.xsd"
+  logicalFilePath="KapuaDB/changelog-certificate-0.3.x.xml">
```
Comment (Coduz) [inline service/security/certificate/internal/src/main/resources/liquibase/0.3.x/crt_certificate.xml]: Version should be 0.4.0  and also other references

File: service/security/certificate/internal/src/main/resources/liquibase/changelog-certificate-master.xml [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+
+    Contributors:
+        Eurotech - initial API and implementation
+ -->
+ <databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog
+                      http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.0.xsd">
+
+    <include relativeToChangelogFile="true" file="./0.3.x/changelog-certificate-0.3.x.xml" />
```
Comment (Coduz) [inline service/security/certificate/internal/src/main/resources/liquibase/changelog-certificate-master.xml]: Version should be 0.4.0

File: pom.xml (hunk 1 above)
Comment (lorthirk) [reply pom.xml]: Of course. I really don't recall of making such change, but I'll fix it anyway.

**Pull request merged**